staking/precompile.go

```diff
@@ -0,0 +1,233 @@
+package staking
+
+import (
+	"bytes"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/harmony-one/harmony/accounts/abi"
+	stakingTypes "github.com/harmony-one/harmony/staking/types"
+	"github.com/pkg/errors"
+)
+
+var abiStaking abi.ABI
+
+func init() {
+	// for commission rates => solidity does not support floats directly
+	// so send commission rates as string
+	StakingABIJSON := `
+	[
+	  {
+	    "inputs": [
```

Comment:
These ABI are all update to date right?

Reply:
Yes, please see [here](https://github.com/MaxMustermann2/harmony-staking-precompiles/blob/main/contracts/lib/StakingPrecompiles.sol) for the traditional staking selectors and [here](https://github.com/MaxMustermann2/harmony-staking-precompiles/blob/main/contracts/lib/MigrationPrecompileSelectors.sol) for the migration precompile.